src/Model/State.re

```diff
@@ -133,3 +133,11 @@ let initial = (~getUserSettings) => {
   terminals: Feature_Terminal.initial,
   textContentProviders: [],
 };
+
+let commands = state =>
+  Command.Lookup.unionMany([
+    Feature_Commands.all(state.commands),
+    Extensions.commands(state.extensions)
+    |> Command.Lookup.fromList
+    |> Command.Lookup.map(msg => Actions.Extension(msg)),
+  ]);
```

Comment:
After way too many iterations trying to fit this somewhere that's both accessible everywhere needed, has all the dependencies line up with both the new and old architecture, and also makes sense, I finally settled on putting it in `State`. I think this will be a pattern I'll use frequently going forward, until we have a proper composition root where things like these can be defined.

Reply:
Sounds good 👍